### PR TITLE
Add App support for Smaato

### PR DESCRIPTION
**Pull request by RodionOrets**
`go` repo PR: https://github.com/prebid/prebid-server/pull/1767

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java [hunk 1]
```diff
@@ -166,6 +168,16 @@ private Site modifySite(Site site, String firstPublisherId) {
         return siteBuilder.build();
     }
 
+    private App modifyApp(App app, String publishedId) {
+        if (app == null) {
+            return null;
+        }
+
+        return app.toBuilder()
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java:176]: We need unit tests for this case, also can be updated integration test

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java [hunk 2]
```diff
@@ -166,6 +168,16 @@ private Site modifySite(Site site, String firstPublisherId) {
...
+                .publisher(Publisher.builder().id(publishedId).build())
+                .build();
+    }
+
```
[the remark this answers is not in the document]
Comment (RodionOrets) [reply src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java:180]: Right. Will do

File: src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java (hunk 1 above)
Comment (RodionOrets) [reply src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java]: Done

File: src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java [hunk 3]
```diff
@@ -184,6 +185,33 @@ public void makeHttpRequestsShouldModifyRequestSite() {
...
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .id("123")
+                        .banner(Banner.builder()
+                                .id("banner_id").format(asList(Format.builder().w(300).h(500).build(),
+                                        Format.builder().w(450).h(150).build())).build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpSmaato.of("publisherId", "adspaceId")))).build()))
+                .app(App.builder().build())
+                .build();
```
Comment (snahornyi) [inline src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java:200]: Can be simplified, if you add `.w(300).h(500)` to givenImp() banner object,
than u can use 
```java
final BidRequest bidRequest = givenBidRequest(bidRequestBuilder ->
                bidRequestBuilder.app(App.builder().build()), identity());
```

File: src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java [hunk 4]
```diff
@@ -184,6 +185,33 @@ public void makeHttpRequestsShouldModifyRequestSite() {
...
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = smaatoBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
```
Comment (snahornyi) [inline src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java:208]: .extracting(HttpRequest::getPayload)

File: src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java [hunk 5]
```diff
@@ -184,6 +185,33 @@ public void makeHttpRequestsShouldModifyRequestSite() {
                 .containsOnly("keywords");
     }
 
+    @Test
+    public void makeHttpRequestsShouldModifyRequestApp() {
```
Comment (snahornyi) [inline src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java:189]: In assert part, u checks only publisher ID, not whole app

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java (hunk 5 above)
Comment (RodionOrets) [reply src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java:189]: That's right. App would contain only `publisherId`: https://github.com/prebid/prebid-server/pull/1767/files#diff-edc0defc783203ca8071cac6a663d13097eed50ab60b283571aa10bffcb95781R104

File: src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java (hunk 4 above)
Comment (RodionOrets) [reply src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java]: Changed

File: src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java (hunk 3 above)
Comment (RodionOrets) [reply src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java]: Done

File: src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java (hunk 5 above)
Comment (snahornyi) [reply src/test/java/org/prebid/server/bidder/smaato/SmaatoBidderTest.java:189]: So makeHttpRequestsShouldModifyRequestAppPublisherId

**Review by snahornyi — CHANGES_REQUESTED** [review]
Pls remove new integration test files, as u can add App object in existing test files

**Review by snahornyi — APPROVED** [review]

**Pull request merged**